tests/src/JIT/Performance/CodeQuality/BenchmarksGame/k-nucleotide/k-nucleotide.cs

```diff
@@ -0,0 +1,284 @@
+﻿/* The Computer Language Benchmarks Game
+   http://benchmarksgame.alioth.debian.org/
+ *
+ * submitted by Josh Goldfoot
+ * 
+ */
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+[assembly: OptimizeForBenchmarks]
+[assembly: MeasureInstructionsRetired]
+[assembly: MeasureGCCounts]
+
+namespace BenchmarksGame
+{
+
+public class knucleotide
+{
+#if DEBUG
+    const int Iterations = 1;
+    const string InputFile = "knucleotide-input.txt"; 
+#else
+    const int Iterations = 10;
+    const string InputFile = "knucleotide-input-big.txt"; 
+#endif
+
+
+    static string FindInput(string s)
+   {
+       string CoreRoot = System.Environment.GetEnvironmentVariable("CORE_ROOT");
+
+       if (CoreRoot == null)
+       {
+           Console.WriteLine("This benchmark requries CORE_ROOT to be set");
+           return null;
+       }
+
+       string inputFile = s ?? InputFile;
+
+       // Normal testing -- input file will end up next to the assembly
+       // and CoreRoot points at the test overlay dir
+       string[] pathPartsNormal = new string[] {
+           CoreRoot, "..", "..", "JIT", "Performance",
+           "CodeQuality", "BenchmarksGame", "k-nucleotide", "k-nucleotide", inputFile
+       };
+
+       string inputPathNormal = Path.Combine(pathPartsNormal);
+
+       // Perf testing -- input file will end up next to the assembly
+       // and CoreRoot points at this directory
+       string[] pathPartsPerf = new string[] { CoreRoot, inputFile };
+
+       string inputPathPerf = Path.Combine(pathPartsPerf);
+
+       string inputPath = null;
+
+       if (File.Exists(inputPathNormal))
+       {
+           inputPath = inputPathNormal;
+       }
+       else if (File.Exists(inputPathPerf))
+       {
+           inputPath = inputPathPerf;
+       }
+
+       if (inputPath != null)
+       {
+           Console.WriteLine("Using input file {0}", inputPath);
+       }
+       else
+       {
+           Console.WriteLine("Unable to find input file {0}", inputFile);
+       }
+
+       return inputPath;
+   }
+
+    public static void Main(string[] args)
```

Comment:
`Main` needs to be `public static int` and return 100 for success, -1 for failure.

Since you're currently not validating the results you can just return 100 from `Main` always. Though it might be nice to have the various `Write` methods return counts and check those against the correct values (which will differ for the different input files)